Graph/Basic Graph Traversal/Backtracking/2096-Step-By-Step_Directions_From_A_Binary_Tree_Node_To_Another.py

```diff
@@ -0,0 +1,50 @@
+# Definition for a binary tree node.
+# class TreeNode:
+#     def __init__(self, val=0, left=None, right=None):
+#         self.val = val
+#         self.left = left
+#         self.right = right
+
+# Graph + Backtracking
+class Solution:
+    def getDirections(self, root: Optional[TreeNode], startValue: int, destValue: int) -> str:
+        self.graph = defaultdict(list)
+        queue = deque([root])
+        self.min_length = float('inf')
+        self.paths = []
+
+        while queue:
+            node = queue.popleft()
+
+            if node.left:
+                self.graph[node.val].append((node.left.val, 'L'))
+                self.graph[node.left.val].append((node.val, 'U'))
+                queue.append(node.left)
+
+            if node.right:
+                self.graph[node.val].append((node.right.val, 'R'))
+                self.graph[node.right.val].append((node.val, 'U'))
+                queue.append(node.right)
+
+        self.backtrack(startValue, destValue, [], set())
+
+        for path in self.paths:
```

Comment:
Since this is a tree, I doubt you will get more than one path from `self.backtrack()`.